index.js

```diff
@@ -102,6 +102,34 @@ module.exports.lintFiles = (patterns, opts) => {
 			});
 		}
 
+		const isTSFile = ext => ext === '.ts' || ext === '.tsx';
+
+		if (paths.find(filePath => isTSFile(path.extname(filePath)))) {
+			opts.parser = 'typescript-eslint-parser';
+
+			const extensions =
```

Comment:
There is probably a better way to do this

Reply:
You should use `mergeWith` like in the rest of the codebase: https://github.com/xojs/xo/blob/master/lib/options-manager.js#L168